src/launch.ts

```diff
@@ -1,360 +1,362 @@
+        // (more details in the comment at the beginning of this function).
+        if (miMode === "lldb" && process.platform === "darwin") {
+            const cpptoolsExtension: vscode.Extension<any> | undefined = vscode.extensions.getExtension('ms-vscode.cpptools');
+            miDebuggerPath = cpptoolsExtension ? path.join(cpptoolsExtension.extensionPath, "debugAdapters", "lldb-mi", "bin", "lldb-mi") : undefined;
+        } else if (miDebuggerPath && miMode) {
+            miDebuggerPath = path.join(miDebuggerPath, miMode);
+            if (process.platform === "win32") {
+                miDebuggerPath = miDebuggerPath + ".exe";
+            }
+        }
+
+        let debugConfig: vscode.DebugConfiguration = {
+            type: dbg,
+            name: `Debug My Program`,
+            request: 'launch',
+            cwd: this.getLaunchTargetDirectory(),
+            args,
+            program: this.getLaunchTargetPath(),
+            MIMode: miMode,
+            miDebuggerPath: miDebuggerPath,
+            console: "internalConsole",
+            internalConsoleOptions: "openOnSessionStart",
+            stopAtEntry: currentLaunchConfiguration.stopAtEntry || defaultLaunchConfiguration?.stopAtEntry,
+            symbolSearchPath: currentLaunchConfiguration.symbolSearchPath || defaultLaunchConfiguration?.symbolSearchPath
+        };
+
+        logger.message("Created the following debug config:\n   type = " + debugConfig.type +
+                       "\n   cwd = " + debugConfig.cwd + " (= " + this.getLaunchTargetDirectory() + ")" +
+                       "\n   args = " + args.join(" ") +
+                       "\n   program = " + debugConfig.program + " (= " + this.getLaunchTargetPath() + ")" +
+                       "\n   MIMode = " + debugConfig.MIMode +
+                       "\n   miDebuggerPath = " + debugConfig.miDebuggerPath +
+                       "\n   stopAtEntry = " + debugConfig.stopAtEntry +
+                       "\n   symbolSearchPath = " + debugConfig.symbolSearchPath);
+
+        return debugConfig;
+    }
+
+    async validateLaunchConfiguration(op: make.Operations): Promise<string> {
+        // Cannot debug the project if it is currently building or (pre-)configuring.
+        if (make.blockedByOp(op)) {
+            return LaunchStatuses.blocked;
+        }
+
+        if (configuration.getBuildBeforeLaunch()) {
+            let currentBuildTarget: string = configuration.getCurrentTarget() || "";
+            logger.message(`Building current target before launch: "${currentBuildTarget}"`);
+            let buildSuccess: boolean = (await make.buildTarget(make.TriggeredBy.buildTarget, currentBuildTarget, false)) === make.ConfigureBuildReturnCodeTypes.success;
+            if (!buildSuccess) {
+                logger.message(`Building target "${currentBuildTarget}" failed.`);
+                let noButton: string = localize("No", "No");

```

Comment:
all the other key names start with a lower-case letter. I recommend that for Yes/No as well.